include/task.h

```diff
@@ -68,6 +68,33 @@ namespace detail {
 		std::unordered_multimap<buffer_id, std::unique_ptr<range_mapper_base>> map;
 	};
 
+	class side_effect_map : private std::unordered_map<host_object_id, experimental::side_effect_order> {
```

Comment:
I'm wondering whether we should simply turn this into an alias for the time being, as it will probably be some time before we'll add additional `side_effect_order`s?

Reply:
`side_effect_order` is currently also unused, but removing it would require us to re-structure the code only to undo that step later. I will keep `side_effect_order` and `side_effect_map` as they are for now.